### PR TITLE
OpenGPG plugin: Don't list expired/revoked GPG key

### LINKED ISSUE #91: Potential social engineering vulnerability in Windows
Unfortunately, I discovered that for Windows I could easily run the executable directly from Dino! 
The only thing that can stop the launch is UAC and then if the program asks for administrator rights.

I think running executable files should be prohibited for security reasons.

### PR DESCRIPTION
**Pull request by vlomovtsev**
This PR implement following changes:

>  - closes #91
>  - Mention that GPG key may be expired or revoked: in the account dialog if the number of OpenPGP keys found is 0, the label also notes that a key may have been revoked or expired.
>  - blocks input in chat box if key is use is revoked or expired

Original changes are picked up from the following one: https://github.com/eerielili/dino/commit/2f3ddad1e87f99cdda9d42dbabc528c2b29c0476

### CODE REVIEW

File: plugins/openpgp/src/gpgme_helper.vala [hunk 1]
```diff
@@ -112,7 +112,13 @@ public static Gee.List<Key> get_keylist(string? pattern = null, bool secret_only
         try {
             while (true) {
                 Key key = context.op_keylist_next();
-                keys.add(key);
+                if (!key.expired && !key.revoked){
```
Comment (mxlgv) [inline plugins/openpgp/src/gpgme_helper.vala:115]: Please fix code style: `if (...) {` and:
``` 
if (...) {
} else {
}
```

File: plugins/openpgp/src/encryption_list_entry.vala [hunk 2]
```diff
@@ -40,6 +41,14 @@ private class EncryptionListEntry : Plugins.EncryptionListEntry, Object {
             return;
         }
 
+        GPG.Key key_check = GPGHelper.get_public_key(db.get_account_key(conversation.account));
+        if (key_check.expired || key_check.revoked){
```
Comment (mxlgv) [inline plugins/openpgp/src/encryption_list_entry.vala:45]: Please fix code style: `if (...) {`

File: plugins/openpgp/src/encryption_list_entry.vala [hunk 3]
```diff
@@ -32,6 +32,7 @@ private class EncryptionListEntry : Plugins.EncryptionListEntry, Object {
         return null;
     }
 
+
```
Comment (mxlgv) [inline plugins/openpgp/src/encryption_list_entry.vala:35]: Unnecessary newline?

**Review by mxlgv — CHANGES_REQUESTED** [review]

**Review by mxlgv — CHANGES_REQUESTED** [review]

File: plugins/openpgp/src/gpgme_helper.vala (hunk 1 above)
Comment (vlomovtsev) [reply plugins/openpgp/src/gpgme_helper.vala]: fixed.

File: plugins/openpgp/src/encryption_list_entry.vala (hunk 2 above)
Comment (vlomovtsev) [reply plugins/openpgp/src/encryption_list_entry.vala]: fixed.

File: plugins/openpgp/src/account_settings_entry.vala [hunk 4]
```diff
@@ -69,7 +69,7 @@ public class AccountSettingsEntry : Plugins.AccountSettingsEntry {
             return;
         }
         if (keys.size == 0) {
-            label.set_markup(build_markup_string(_("Key publishing disabled"), _("No keys available. Generate one!")));
+            label.set_markup(build_markup_string(_("Key publishing disabled"), _("No keys available. Generate one or check if your keys aren't expired or revoked!")));
```
[the remark this answers is not in the document]
Comment (vlomovtsev) [reply plugins/openpgp/src/account_settings_entry.vala]: moved extended message to specific language file, for now  only for en/ru localizations.

**Review by giantplaceholder — APPROVED** [review]

**Comment by eerielili** [comment]
@mxlgv 
If interested : I did update for new account settings UI, see https://github.com/dino/dino/pull/1540

**Pull request merged**